spec/javascripts/controllers/ems_common/arbitration_profile_form_controller_spec.js

```diff
@@ -0,0 +1,131 @@
+describe('arbitrationProfileFormController', function() {
+  var $scope, $location, $controller, $httpBackend, miqService, postService;
+
+  beforeEach(module('ManageIQ'));
+
+  beforeEach(inject(function($rootScope, $location, _$controller_, _$httpBackend_, _miqService_, _postService_, _API_) {
+    miqService = _miqService_;
+    postService = _postService_;
+    spyOn(miqService, 'showButtons');
+    spyOn(miqService, 'hideButtons');
+    spyOn(miqService, 'miqAjaxButton');
+    spyOn(miqService, 'sparkleOn');
+    spyOn(miqService, 'sparkleOff');
+    spyOn(postService, 'cancelOperation');
+    spyOn(postService, 'saveRecord');
+    spyOn(postService, 'createRecord');
+    $scope = $rootScope.$new();
+
+    spyOn($location, 'absUrl').and.returnValue('/ems_cloud/arbitration_profile_edit/1000000000001?db=ems_cloud')
+    spyOn(window, 'queryParam').and.returnValue(1000000000001)
+
+    $location = $location;
+    $httpBackend = _$httpBackend_;
+
+    var arbitrationProfileData = {name:                 'arbitrationProfileName',
+                                  description:          'arbitrationProfileDescription',
+                                  ems_id:               '1000000000001',
+                                  authentication_id:    '1000000000005',
+                                  availability_zone_id: '1000000000006',
+                                  cloud_network_id:     '1000000000007',
+                                  cloud_subnet_id:      '1000000000008',
+                                  flavor_id:            '1000000000009',
+                                  security_group_id:    ''
+    };
+
+    var arbitrationProfileOptions = {availability_zones: []};
+    spyOn(_API_, 'get').and.returnValue(Promise.resolve(arbitrationProfileData))
+
+    $controller = _$controller_('arbitrationProfileFormController', {
+      $scope: $scope,
+      $location: $location,
+      arbitrationProfileFormId: 1000000000001,
+      miqService: miqService,
+      arbitrationProfileData: arbitrationProfileData,
+      arbitrationProfileOptions: arbitrationProfileOptions
```

Comment:
aaah, you may need to `API: _API_,` here for it to work :)

Also, `bower install -S qs` (or --save) so that it changes bower.json, otherwise travis errors out..